telegram-bot-api/Client.cpp

```diff
@@ -8638,6 +8733,66 @@ td::Status Client::process_get_memory_stats_query(PromisedQueryPtr &query) {
   */
   return Status::OK();
 }
+
+td::Status Client::process_get_proxies_query(PromisedQueryPtr &query) {
+  send_request(make_object<td_api::getProxies>(),
+               std::make_unique<TdOnGetProxiesQueryCallback>(std::move(query)));
+  return Status::OK();
+}
+
+td::Status Client::process_add_proxy_query(PromisedQueryPtr &query) {
+  TRY_RESULT(server, get_required_string_arg(query.get(), "server"));
+  TRY_RESULT(proxy_type, get_required_string_arg(query.get(), "type"));
+  td::int32 port = get_integer_arg(query.get(), "port", 0, 0, 65535);
+  if (!port) {
+    return Status::Error(400, PSLICE() << "Parameter port with a value between 1-65535 is required");
+  }
+
+  td_api::object_ptr<td_api::ProxyType> type;
+  td::to_lower_inplace(proxy_type);
+  if (proxy_type == "mtproto") {
+    TRY_RESULT(proxy_secret, get_required_string_arg(query.get(), "secret"));
+    type = td_api::make_object<td_api::proxyTypeMtproto>(proxy_secret.str());
+  } else if (proxy_type == "socks5") {
+    auto proxy_user = query->arg("username");
+    auto proxy_pass = query->arg("password");
+    type = td_api::make_object<td_api::proxyTypeSocks5>(proxy_user.str(), proxy_pass.str());
+  } else {
```

Comment:
you're missing https://core.telegram.org/tdlib/docs/classtd_1_1td__api_1_1proxy_type_http.html